tests/functional/api/cursor_api_test.c

```diff
@@ -3,15 +3,25 @@
  * Copyright (C) 2021 Micron Technology, Inc.  All rights reserved.
```

Comment:
I firmly believe we should document each API test case (I would like to see all test cases documented, but let's start with the API tests).  See the following example for cursor create.  We don't need to use this particular format (I might prefer annotating the code instead of a block comment so it's clear what code is implementing what test).
```
// Cursor create test cases
// ---------------------------
// API:
//     hse_kvs_cursor_create(
//         struct hse_kvs         *kvs,
//         const unsigned int      flags,
//         struct hse_kvdb_txn    *txn,
//         const void             *pfx,
//         size_t                  pfx_len,
//         struct hse_kvs_cursor **cursor);
//
// In the following test cases, unless otherwise stated: kvs is a
// non-txn KVS, flags==0, txn==NULL, pfx==NULL, pfx_len==0, and cursor
// is valid pointer.
//
// Test cases:
// - Invalid param: kvs == NULL
// - Invalid param: flags == -1
// - Invalid param: pfx != NULL && pfx_len == 0  (TBD: maybe this is valid?)
// - Invalid param: pfx == NULL && pfx_len > 0
// - Invalid param: cursor == NULL
// - Too many active cursors (need backdoor API to get this limit?)
// - txn=NULL on txn KVS --> sucess
// - txn=NULL on non-txn KVS --> success
// - With active txn on non-txn KVS --> EINVAL
// - With inactive txn on non-txn KVS --> EINVAL
// - With active txn on a txn KVS --> success
// - With inactive txn on txn KVS --> EPROTO
//
// Repeat the success cases from above with a reverse flag,
// with a prefix key, and with both:
//
// - With reverse_flag:
//   - txn=NULL on txn KVS --> sucess
//   - txn=NULL on non-txn KVS --> success
//   - With active txn on a txn KVS --> success
//
// - With valid prefix key:
//   - txn=NULL on txn KVS --> sucess
//   - txn=NULL on non-txn KVS --> success
//   - With active txn on a txn KVS --> success
//
// - With reverse_flag and valid prefix key:
//   - txn=NULL on txn KVS --> sucess
//   - txn=NULL on non-txn KVS --> success
//   - With active txn on a txn KVS --> success
//
// Not tested:
// - failures in kvs_cursor_alloc (low memory, cursor cache issues)
// - failures in acquiring a viewset
```

Reply:
To me the test names feel obvious what each test is looking for. Will evaluate after I am done. Need to continue slogging through the implementations of the tests. Annoying after a while. One man can only type `MTF` so many times.

Reply:
Seems like cursor create could have more test cases.  This PR has the following cases  for cursor create:
```
create_null_kvs
create_invalid_flags
create_null_cursor
```
But I've identified about 20 test cases in the above comment.

Reply:
Yep seems like the ball was dropped there. Let me continue to work on this.